js/app/controllers/calcontroller.js

```diff
@@ -140,17 +140,28 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
-				is.loading = false;
-				// TODO - scope.apply should not be necessary here
-				$scope.$apply();
-			}).catch((reason) => {
-				angular.element('#header-right').css('display', 'none');
-				angular.element('#emptycontent-container').css('display', 'block');
-			});
+		    $scope.calendars = [];
+		    var tokens = constants.publicSharingToken.split(".");
+		    var finished = 0;
+		    tokens.forEach(function(token){
+                $scope.calendarsPromise = CalendarService.getPublicCalendar(token).then(function(calendar) {
+                    finished++;
+                    $scope.calendars.push(calendar);
+                    if(finished >= tokens.length){
+                        is.loading = false;
+                        // TODO - scope.apply should not be necessary here
+                        $scope.$apply();
+                    }
+                }).catch((reason) => {
+                    angular.element('#header-right').css('display', 'none');
+                    angular.element('#emptycontent-container').css('display', 'block');
+                });
+            });
 		}
 
+        $scope.integration = function () {
+            return '<iframe width="400" height="215" src="' + $scope.$parent.root + 'embed/' + constants.publicSharingToken + '"></iframe>';
```

Comment:
Not sure if angular handles escaping here.
What if you pass html as a parameter for `constants.publicSharingToken`?

This might result in an html injection.

Can you please check if its properly escaped?

Reply:
Will check